src/main/java/com/nekomaster1000/infernalexp/events/MiscEvents.java

```diff
@@ -75,25 +79,50 @@ public static void onModConfigEvent(final ModConfig.ModConfigEvent event) {
     //Blocks being broken
     @SubscribeEvent
     public void onBlockBreak(BlockEvent.BreakEvent event) {
-        if (event.getState().equals(Blocks.CRIMSON_FUNGUS.getDefaultState())
-            || event.getState().equals(Blocks.CRIMSON_ROOTS.getDefaultState())
-            || event.getState().equals(Blocks.CRIMSON_STEM.getDefaultState())
-            || event.getState().equals(Blocks.STRIPPED_CRIMSON_STEM.getDefaultState())
-            || event.getState().equals(Blocks.WEEPING_VINES.getDefaultState())
-            || event.getState().equals(Blocks.WEEPING_VINES_PLANT.getDefaultState())
-            || event.getState().equals(Blocks.NETHER_WART_BLOCK.getDefaultState())) {
-            List<?> list = event.getPlayer().world.getEntitiesWithinAABB(ShroomloinEntity.class,
-                event.getPlayer().getBoundingBox().grow(32.0D));
-            for (int j = 0; j < list.size(); j++) {
-                Entity entity = (Entity) list.get(j);
-                if (entity instanceof ShroomloinEntity) {
-                    ShroomloinEntity shroomloinEntity = (ShroomloinEntity) entity;
-                    shroomloinEntity.becomeAngryAt(event.getPlayer());
+        BlockState state = event.getState();
+        List<?> list = event.getPlayer().world.getEntitiesWithinAABB(ShroomloinEntity.class,
+            event.getPlayer().getBoundingBox().grow(32.0D));
+        for (int j = 0; j < list.size(); j++) {
+            Entity entity = (Entity) list.get(j);
+            if (entity instanceof ShroomloinEntity) {
+                ShroomloinEntity shroomloinEntity = (ShroomloinEntity) entity;
+
+                if (((ShroomloinEntity) entity).getFungusType() == 1) {
+                    if (state.getBlock().isIn(IETags.Blocks.ANGER_CRIMSON_BLOCKS)) {
```

Comment:
Could you please refactor these tags to be ANGER_CRIMSON_SHROOMLOIN_BLOCKS, etc.